src/NuGet.Core/NuGet.Packaging/Core/comparers/PackageDependencyGroupComparer.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using NuGet.Frameworks;
+using NuGet.Shared;
+
+namespace NuGet.Packaging.Core
+{
+    public class PackageDependencyGroupComparer : IEqualityComparer<PackageDependencyGroup>
```

Comment:
Personally, I don't feel too strongly about having a separate comparer vs implementing in the class, but I don't think this *has* to be a public comparer. 

https://github.com/NuGet/NuGet.Client/blob/dev/docs/nuget-sdk.md#changes-to-apis

IMO, this qualifies under, does it need to be public?